pkg/backend/display/object_diff.go

```diff
@@ -1215,13 +1220,23 @@ func (p *propertyPrinter) decodeValue(repr string) (resource.PropertyValue, stri
 			return nil, "", false
 		}
 
-		r.Reset(repr)
-		if err := yaml.NewDecoder(r).Decode(&object); err == nil {
-			translated, ok := p.translateYAMLValue(object)
-			if !ok {
-				return nil, "", false
+		pos, err := r.Seek(0, io.SeekStart)
+		contract.Assertf(err == nil, "strings.NewReader.Seek(0, io.SeekStart) failed: %v", err)
+		contract.Assertf(pos == 0, "strings.NewReader.Seek(0, io.SeekStart) failed: pos=%d", pos)
+
+		yamlDecoder := yaml.NewDecoder(r)
+		if err := yamlDecoder.Decode(&object); err == nil {
+			// Make sure _all_ the string was consumed as YAML. Unlike JsonDecoder above, the YamlDecoder
```

Comment:
Could we not check the position of the reader?
https://pkg.go.dev/strings#Reader.Len returns the remaining unread bytes.
(Although this won't do anything if the YAMl decoder reads ahead and buffers.)

Reply:
I did check this but yeh the YAML decoder reads ahead